paimon-core/src/main/java/org/apache/paimon/io/KeyValueFileWriterFactory.java

```diff
@@ -164,11 +165,14 @@ private Builder(
         public KeyValueFileWriterFactory build(
                 BinaryRow partition, int bucket, CoreOptions options) {
             RowType fileRowType = KeyValue.schema(keyType, valueType);
+            RowType storageRowType = RowTypeUtils.toStorageRowType(fileRowType);
```

Comment:
Here we need an option, by default, we can use old style row type. Because we need to be as compatible with old readers as possible.

Then wait for 1-2 versions, and switch to the default mode to thin mode.